### PR TITLE
Add support for accessing linking objects / backlinks

### PR DESCRIPTION
**Pull request by mrackwitz**
Coming from the Cocoa SDKs, I was missing backlinks when working with the JS SDK, so I dove in my spare time a little into the codebase and wanted to see how it could be done.

This is related to #343.

/c @kristiandupont @bmunkholm

### CODE REVIEW

**Comment by ianpward** [comment]
Excellent!

File: src/js_schema.hpp [hunk 1]
```diff
@@ -177,14 +187,27 @@ ObjectSchema Schema<T>::parse_object_schema(ContextType ctx, ObjectType object_s
         for (uint32_t i = 0; i < length; i++) {
             ObjectType property_object = Object::validated_get_object(ctx, properties_object, i);
             std::string property_name = Object::validated_get_string(ctx, property_object, name_string);
-            object_schema.persisted_properties.emplace_back(parse_property(ctx, property_object, property_name, object_defaults));
+            Property property = parse_property(ctx, property_object, property_name, object_defaults);
+            if (property.type == realm::PropertyType::LinkingObjects) {
+                object_schema.computed_properties.emplace_back(property);
```
Comment (fealebenpae) [inline src/js_schema.hpp]: We should `std::move` the property.

File: tests/js/linkingobjects-tests.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2016 Realm Inc.
```
Comment (fealebenpae) [inline tests/js/linkingobjects-tests.js]: Should be `2017`

**Comment by fealebenpae** [comment]
There seem to be some test failures in the Chrome debugger. Could be related to #1096. I'll investigate.

I'll also add API docs.

**Comment by astigsen** [comment]
In js it is super usual to open the realms without specifying a schema (since it can be inferred from the file itself), so LinkingProperties should also work dynamically without a predefined schema.

**Pull request merged**